.github/workflows/code-quality.yml

```diff
@@ -0,0 +1,21 @@
+name: Code Quality Checks
+on: [pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Checks
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -e .
+        pip install black==23.1.0 pylint==v2.16.2
+    - name: Format check
+      run: black --diff --check $(git ls-files 'astarte')
```

Comment:
I suggest setting the number of lines of code for black to `100` to keep it consistent with other SDK such as Rust.